### PR TITLE
feat: add share helpers

### PR DESCRIPTION
**Pull request by gpsanant**
added a view func for converting deposit shares from withdrawable shares

### CODE REVIEW

File: src/contracts/core/DelegationManager.sol [hunk 1]
```diff
@@ -970,6 +970,21 @@ contract DelegationManager is
         }
     }
 
+    /// @inheritdoc IDelegationManager
+    function convertToDepositShares(address staker, IStrategy[] memory strategies, uint256[] memory shares) external view returns (uint256[] memory) {
```
Comment (8sunyuan) [inline src/contracts/core/DelegationManager.sol:974]: nit: use withdrawable shares as input

File: src/test/unit/DelegationUnit.t.sol [hunk 2]
```diff
@@ -8274,3 +8274,98 @@ contract DelegationManagerUnitTests_Lifecycle is DelegationManagerUnitTests {
...
+            // so this value is sane to input into `completeQueuedWithdrawals`
+            assertLe(
+                depositShares[i],
+                expectedDepositShares[i],
+                "deposit shares should be less than or equal to expected deposit shares"
+            );
+        }
+
+        // get the deposit shares
+        uint256[] memory oneThirdWithdrawableShares = new uint256[](strategies.length);
```
Comment (8sunyuan) [inline src/test/unit/DelegationUnit.t.sol:8356]: Not quite sure this is being done here in the helper

Comment (gpsanant) [reply src/test/unit/DelegationUnit.t.sol:8356]: it's just to make sure the thing properly scales down as well when called with smaller values that the max amount of shares

File: src/contracts/core/DelegationManager.sol (hunk 1 above)
Comment (gpsanant) [reply src/contracts/core/DelegationManager.sol]: https://github.com/Layr-Labs/eigenlayer-contracts/pull/964/commits/36e8a850306ad710f3ac7499e1baccbd49a81d8b

**Review by 8sunyuan — APPROVED** [review]

**Review by ypatil12 — APPROVED** [review]

**Pull request merged**